### ISSUE TITLE
[Bug]: Full-text search does not support iterator search, error: unsupported metrics type for search iteration: BM25

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#36966]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20750c0-20241017
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```

[2024-10-17 17:44:53 - INFO - ci_test]: ################################################################################ (conftest.py:232)
[2024-10-17 17:44:53 - INFO - ci_test]: [initialize_milvus] Log cleaned up, start testing... (conftest.py:233)
[2024-10-17 17:44:53 - INFO - ci_test]: [setup_class] Start setup class... (client_base.py:40)
[2024-10-17 17:44:53 - INFO - ci_test]: *********************************** setup *********************************** (client_base.py:46)
[2024-10-17 17:44:53 - INFO - ci_test]: pymilvus version: 2.5.0rc96 (client_base.py:47)
[2024-10-17 17:44:53 - INFO - ci_test]: [setup_method] Start setup test case test_full_text_search_with_search_iterator. (client_base.py:49)
-------------------------------- live log call ---------------------------------
[2024-10-17 17:44:53 - INFO - ci_test]: server version: 20750c0-dev (client_base.py:165)
[2024-10-17 17:44:54 - INFO - ci_test]: dataframe
        id       word                                           sentence                                          paragraph                                               text                                                emb
0        0     career                                      them at some.  stage us ok office sit rate. think cold marria...  election new risk along. start admit parent be...  [0.15024196913948495, 0.7622556594314852, 0.00...
1        1       same                       whose idea expect party far.  nothing water bank full close. drop strong fiv...  could popular world clearly lot. method star o...  [0.7614539018042723, 0.10171613691234949, 0.58...
2        2    mention  try offer citizen because discuss station arti...  three order rather network fund none. owner co...  month something their. all side focus once onl...  [0.8285398604112688, 0.00545147517579736, 0.47...
3        3  necessary        discuss share month establish they account.  day financial red ahead watch design. notice r...  special moment fire loss best pick. mr full pl...  [0.919742438405048, 0.49461467621789923, 0.114...
4        4       that                       account guess live continue.  worry page night design. discussion will road ...  field full include five middle goal. specific ...  [0.9069278550569845, 0.23919210238906297, 0.01...
...    ...        ...                                                ...                                                ...                                                ...                                                ...
4995  4995    mention  method finish show present of money everything...  none keep stage at him others herself enjoy. c...  say traditional view term. per admit ability e...  [0.09731482606334496, 0.26550716673116537, 0.9...
4996  4996  agreement                       continue probably per class.  season structure pull defense concern pay figu...  happen what guess and personal year three. fou...  [0.5296512360997506, 0.8756720189229045, 0.588...
4997  4997       game       mrs trial choice evening economy first drug.  word value nation past race have happen. toget...  force go along represent skin. meet threat fly...  [0.38255253103438225, 0.4156154930465835, 0.38...
4998  4998       read                    mean image western detail also.  agent night skill our boy. down real power ite...  themselves writer themselves list realize appr...  [0.9166120826478567, 0.5525119114934106, 0.306...
4999  4999    country  type conference become career value sense scor...  hundred matter tend ground anyone guy now baby...  pass adult effect school while benefit east he...  [0.8357687411558864, 0.7815092628050222, 0.398...

[5000 rows x 6 columns] (test_full_text_search.py:2549)
[2024-10-17 17:44:55 - INFO - ci_test]: Analyze document cost time: 0.06827807426452637 (common_func.py:340)
[2024-10-17 17:46:49 - INFO - ci_test]: search data: ['situation cause help very hotel foot. energy alone subject early public kind close. question feeling several difference bank final.\ndown assume these. itself resource ready lose.life'] (test_full_text_search.py:2584)
[2024-10-17 17:46:49 - ERROR - ci_test]: Traceback (most recent call last):
  File "/Users/zilliz/workspace/milvus/tests/python_client/utils/api_request.py", line 32, in inner_wrapper
    res = func(*args, **_kwargs)
  File "/Users/zilliz/workspace/milvus/tests/python_client/utils/api_request.py", line 63, in api_request
    return func(*arg, **kwargs)
  File "/Users/zilliz/opt/anaconda3/envs/full_text_search/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 976, in search_iterator
    return SearchIterator(
  File "/Users/zilliz/opt/anaconda3/envs/full_text_search/lib/python3.8/site-packages/pymilvus/orm/iterator.py", line 439, in __init__
    self.__init_search_iterator()
  File "/Users/zilliz/opt/anaconda3/envs/full_text_search/lib/python3.8/site-packages/pymilvus/orm/iterator.py", line 458, in __init_search_iterator
    self.__set_up_range_parameters(init_page)
  File "/Users/zilliz/opt/anaconda3/envs/full_text_search/lib/python3.8/site-packages/pymilvus/orm/iterator.py", line 473, in __set_up_range_parameters
    self.__update_width(page)
  File "/Users/zilliz/opt/anaconda3/envs/full_text_search/lib/python3.8/site-packages/pymilvus/orm/iterator.py", line 464, in __update_width
    if metrics_positive_related(self._param[METRIC_TYPE]):
  File "/Users/zilliz/opt/anaconda3/envs/full_text_search/lib/python3.8/site-packages/pymilvus/orm/iterator.py", line 326, in metrics_positive_related
    raise MilvusException(message=f"unsupported metrics type for search iteration: {metrics}")
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=unsupported metrics type for search iteration: BM25)>
 (api_request.py:45)
[2024-10-17 17:46:49 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=unsupported metrics type for search iteration: BM25)> (api_request.py:46)
FAILED
testcases/test_full_text_search.py:2458 (TestSearchWithFullTextSearch.test_full_text_search_with_search_iterator[default-None-SPARSE_INVERTED_INDEX-True-True-0-1])
self = <test_full_text_search.TestSearchWithFullTextSearch object at 0x13151cbe0>
tokenizer = 'default', expr = None, enable_inverted_index = True
enable_partition_key = True, empty_percent = 0
index_type = 'SPARSE_INVERTED_INDEX', nq = 1

    @pytest.mark.tags(CaseLabel.L0)
    @pytest.mark.parametrize("nq", [1])
    @pytest.mark.parametrize("empty_percent", [0])
    @pytest.mark.parametrize("enable_partition_key", [True])
    @pytest.mark.parametrize("enable_inverted_index", [True])
    @pytest.mark.parametrize("index_type", ["SPARSE_INVERTED_INDEX"])
    @pytest.mark.parametrize("expr", [None])
    @pytest.mark.parametrize("tokenizer", ["default"])
    def test_full_text_search_with_search_iterator(
            self, tokenizer, expr, enable_inverted_index, enable_partition_key, empty_percent, index_type, nq
    ):
        """
        target: test full text search
        method: 1. enable full text search and insert data with varchar
                2. search with text
                3. verify the result
        expected: full text search successfully and result is correct
        """
        tokenizer_params = {
            "tokenizer": tokenizer,
        }
        dim = 128
        fields = [
            FieldSchema(name="id", dtype=DataType.INT64, is_primary=True),
            FieldSchema(
                name="word",
                dtype=DataType.VARCHAR,
                max_length=65535,
                enable_tokenizer=True,
                tokenizer_params=tokenizer_params,
                is_partition_key=enable_partition_key,
            ),
            FieldSchema(
                name="sentence",
                dtype=DataType.VARCHAR,
                max_length=65535,
                enable_tokenizer=True,
                tokenizer_params=tokenizer_params,
            ),
            FieldSchema(
                name="paragraph",
                dtype=DataType.VARCHAR,
                max_length=65535,
                enable_tokenizer=True,
                tokenizer_params=tokenizer_params,
            ),
            FieldSchema(
                name="text",
                dtype=DataType.VARCHAR,
                max_length=65535,
                enable_tokenizer=True,
                enable_match=True,
                tokenizer_params=tokenizer_params,
            ),
            FieldSchema(name="emb", dtype=DataType.FLOAT_VECTOR, dim=dim),
            FieldSchema(name="text_sparse_emb", dtype=DataType.SPARSE_FLOAT_VECTOR),
        ]
        schema = CollectionSchema(fields=fields, description="test collection")
        bm25_function = Function(
            name="text_bm25_emb",
            function_type=FunctionType.BM25,
            input_field_names=["text"],
            output_field_names=["text_sparse_emb"],
            params={},
        )
        schema.add_function(bm25_function)
        data_size = 5000
        collection_w = self.init_collection_wrap(
            name=cf.gen_unique_str(prefix), schema=schema
        )
        fake = fake_en
        if tokenizer == "jieba":
            language = "zh"
            fake = fake_zh
        else:
            language = "en"
    
        data = [
            {
                "id": i,
                "word": fake.word().lower() if random.random() >= empty_percent else "",
                "sentence": fake.sentence().lower() if random.random() >= empty_percent else "",
                "paragraph": fake.paragraph().lower() if random.random() >= empty_percent else "",
                "text": fake.text().lower() if random.random() >= empty_percent else "",
                "emb": [random.random() for _ in range(dim)],
            }
            for i in range(data_size)
        ]
        df = pd.DataFrame(data)
        corpus = df["text"].to_list()
        log.info(f"dataframe\n{df}")
        texts = df["text"].to_list()
        word_freq = cf.analyze_documents(texts, language=language)
        tokens = list(word_freq.keys())
        if len(tokens) == 0:
            log.info(f"empty tokens, add a dummy token")
            tokens = ["dummy"]
        batch_size = 5000
        for i in range(0, len(df), batch_size):
            collection_w.insert(
                data[i : i + batch_size]
                if i + batch_size < len(df)
                else data[i : len(df)]
            )
            collection_w.flush()
        collection_w.create_index(
            "emb",
            {"index_type": "HNSW", "metric_type": "L2", "params": {"M": 16, "efConstruction": 500}},
        )
        collection_w.create_index(
            "text_sparse_emb",
            {
                "index_type": index_type,
                "metric_type": "BM25",
                "params": {
                    "bm25_k1": 1.5,
                    "bm25_b": 0.75,
                }
            }
        )
        if enable_inverted_index:
            collection_w.create_index("text", {"index_type": "INVERTED"})
        collection_w.load()
        limit = 1000
        search_data = [fake.text().lower()+random.choice(tokens) for _ in range(nq)]
        log.info(f"search data: {search_data}")
        # get distance with search data
>       iterator, _ = collection_w.search_iterator(
            data=search_data,
            anns_field="text_sparse_emb",
            batch_size=100,
            param={
                "metric_type": "BM25",
            },
            output_fields=["id", "text", "text_sparse_emb"],
            limit=1000
        )
```

**Comment by zhuwenxing** [comment]
/assign @zhengbuqian 

PTAL

**Comment by zhuwenxing** [comment]
verified and fixed in `master-346510e-20241021` and `pymilvus==2.5.0rc101`
